File: src/Entity/Evenement.php
<?php

namespace App\Entity;

use App\Repository\EvenementRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=EvenementRepository::class)
 */
class Evenement
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     */
    private $nom;

    /**
     * @ORM\Column(type="date")
     * @Assert\NotBlank
     */
    private $date_debut;

    /**
     * @ORM\Column(type="date")
     * @Assert\NotBlank
     */
    private $date_fin;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     */
    private $description;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     */
    private $pays;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     */
    private $ville;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $image;

    /**
     * @ORM\Column(type="float")
     * @Assert\NotBlank
     */
    private $prix;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank
     */
    private $nbr_places;

    /**
     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="evenements")
     * @ORM\JoinColumn(nullable=false)
     */
    private $user;

    /**
     * @ORM\OneToMany(targetEntity=Offre::class, mappedBy="evenement", orphanRemoval=true)
     * @Assert\NotBlank
     */
    private $offres;

    /**
     * @ORM\OneToMany(targetEntity=Reservation::class, mappedBy="evenement", orphanRemoval=true)
     * @Assert\NotBlank
     */
    private $reservations;

    /**
     * @ORM\Column(type="float", nullable=true)
     */
    private $rate;

    /**
     * @ORM\Column(type="integer", nullable=true)
     */
    private $vote;

    /**
     * @ORM\OneToMany(targetEntity=EvenementCommentaire::class, mappedBy="event")
     */
    private $evenementCommentaires;

    /**
     * @ORM\OneToMany(targetEntity=EventLike::class, mappedBy="event")
     */
    private $ManyToOne;

    public function __construct()
    {
        $this->offres = new ArrayCollection();
        $this->reservations = new ArrayCollection();
        $this->evenementCommentaires = new ArrayCollection();
        $this->ManyToOne = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(?string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getDateDebut(): ?\DateTimeInterface
    {
        return $this->date_debut;
    }

    public function setDateDebut(?\DateTimeInterface $date_debut): self
    {
        $this->date_debut = $date_debut;

        return $this;
    }

    public function getDateFin(): ?\DateTimeInterface
    {
        return $this->date_fin;
    }

    public function setDateFin(?\DateTimeInterface $date_fin): self
    {
        $this->date_fin = $date_fin;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getPays(): ?string
    {
        return $this->pays;
    }

    public function setPays(?string $pays): self
    {
        $this->pays = $pays;

        return $this;
    }

    public function getVille(): ?string
    {
        return $this->ville;
    }

    public function setVille(?string $ville): self
    {
        $this->ville = $ville;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(?float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getNbrPlaces(): ?int
    {
        return $this->nbr_places;
    }

    public function setNbrPlaces(?int $nbr_places): self
    {
        $this->nbr_places = $nbr_places;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    /**
     * @return Collection|Offre[]
     */
    public function getOffres(): Collection
    {
        return $this->offres;
    }

    public function addOffre(Offre $offre): self
    {
        if (!$this->offres->contains($offre)) {
            $this->offres[] = $offre;
            $offre->setEvenement($this);
        }

        return $this;
    }

    public function removeOffre(Offre $offre): self
    {
        if ($this->offres->removeElement($offre)) {
            // set the owning side to null (unless already changed)
            if ($offre->getEvenement() === $this) {
                $offre->setEvenement(null);
            }
        }

        return $this;
    }

    /**
     * @return Collection|Reservation[]
     */
    public function getReservations(): Collection
    {
        return $this->reservations;
    }

    public function addReservation(Reservation $reservation): self
    {
        if (!$this->reservations->contains($reservation)) {
            $this->reservations[] = $reservation;
            $reservation->setEvenement($this);
        }

        return $this;
    }

    public function removeReservation(Reservation $reservation): self
    {
        if ($this->reservations->removeElement($reservation)) {
            // set the owning side to null (unless already changed)
            if ($reservation->getEvenement() === $this) {
                $reservation->setEvenement(null);
            }
        }

        return $this;
    }

    public function __toString()
    {
        return $this->nom;
    }

    public function getRate(): ?float
    {
        return $this->rate;
    }

    public function setRate(?float $rate): self
    {
        $this->rate = $rate;

        return $this;
    }

    public function getVote(): ?int
    {
        return $this->vote;
    }

    public function setVote(?int $vote): self
    {
        $this->vote = $vote;

        return $this;
    }

    /**
     * @return Collection|EvenementCommentaire[]
     */
    public function getEvenementCommentaires(): Collection
    {
        return $this->evenementCommentaires;
    }

    public function addEvenementCommentaire(EvenementCommentaire $evenementCommentaire): self
    {
        if (!$this->evenementCommentaires->contains($evenementCommentaire)) {
            $this->evenementCommentaires[] = $evenementCommentaire;
            $evenementCommentaire->setEvent($this);
        }

        return $this;
    }

    public function removeEvenementCommentaire(EvenementCommentaire $evenementCommentaire): self
    {
        if ($this->evenementCommentaires->removeElement($evenementCommentaire)) {
            // set the owning side to null (unless already changed)
            if ($evenementCommentaire->getEvent() === $this) {
                $evenementCommentaire->setEvent(null);
            }
        }

        return $this;
    }

    /**
     * @return Collection|EventLike[]
     */
    public function getManyToOne(): Collection
    {
        return $this->ManyToOne;
    }

    public function addManyToOne(EventLike $manyToOne): self
    {
        if (!$this->ManyToOne->contains($manyToOne)) {
            $this->ManyToOne[] = $manyToOne;
            $manyToOne->setEvent($this);
        }

        return $this;
    }

    public function removeManyToOne(EventLike $manyToOne): self
    {
        if ($this->ManyToOne->removeElement($manyToOne)) {
            // set the owning side to null (unless already changed)
            if ($manyToOne->getEvent() === $this) {
                $manyToOne->setEvent(null);
            }
        }

        return $this;
    }

    /**
     * @return boolean
     */
    public function islikebyuser(User $user): bool
    {
        foreach ($this->ManyToOne as $like) {
            if ($like->getUser() === $user) {
                return true;
            }
        }
        return false;
    }
}

File: src/GestionEvents/entites/Evenement.java
package GestionEvents.entites;

import java.sql.Date;

/**
 *
 * @author legion
 */
public class Evenement {
    private int id;
    private int user_id;
    private String nom;
    private Date date_debut;
    private Date date_fin;
    private String description;
    private String pays;
    private String ville;
    private String image;
    private double prix;
    private int nbr_places;
    private double rate;
    private int vote;
    private String username;
    

    public Evenement() {
    }
    
    public Evenement(String nom,Date date_debut,Date date_fin,String description,String pays,String ville,double prix,int nbr_places) {
        this.nom = nom;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description = description;
        this.pays = pays;
        this.ville = ville;
        this.prix = prix;
        this.nbr_places = nbr_places;
    }
    
    public Evenement(int id, int user_id, String nom, Date date_debut, Date date_fin, String description, String pays, String ville, String image, double prix, int nbr_places, double rate, int vote) {
        this.id = id;
        this.user_id = user_id;
        this.nom = nom;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description = description;
        this.pays = pays;
        this.ville = ville;
        this.image = image;
        this.prix = prix;
        this.nbr_places = nbr_places;
        this.rate = rate;
        this.vote = vote;
    }
    
    public Evenement(int user_id, String nom, Date date_debut, Date date_fin, String description, String pays, String ville, String image, double prix, int nbr_places) {
        this.user_id = user_id;
        this.nom = nom;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description = description;
        this.pays = pays;
        this.ville = ville;
        this.image = image;
        this.prix = prix;
        this.nbr_places = nbr_places;
    }

    public Evenement(int id,String username, String nom, Date date_debut, Date date_fin, String description, String pays, String ville, double prix, int nbr_places, double rate) {
        this.id=id;
        this.nom = nom;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description = description;
        this.pays = pays;
        this.ville = ville;
        this.prix = prix;
        this.nbr_places = nbr_places;
        this.rate = rate;
        this.username = username;
    }
    
    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUser_id() {
        return user_id;
    }

    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public Date getDate_debut() {
        return date_debut;
    }

    public void setDate_debut(Date date_debut) {
        this.date_debut = date_debut;
    }

    public Date getDate_fin() {
        return date_fin;
    }

    public void setDate_fin(Date date_fin) {
        this.date_fin = date_fin;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getPays() {
        return pays;
    }

    public void setPays(String pays) {
        this.pays = pays;
    }

    public String getVille() {
        return ville;
    }

    public void setVille(String ville) {
        this.ville = ville;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public int getNbr_places() {
        return nbr_places;
    }

    public void setNbr_places(int nbr_places) {
        this.nbr_places = nbr_places;
    }

    public double getRate() {
        return rate;
    }

    public void setRate(double rate) {
        this.rate = rate;
    }

    public int getVote() {
        return vote;
    }

    public void setVote(int vote) {
        this.vote = vote;
    }
    
    
    
    
    
}
